ui-tests/cypress/integration/content/templates.spec.js

```diff
@@ -0,0 +1,91 @@
+import HomePage from '../../support/pageObjects/HomePage';
+import { htmlElements } from '../../support/pageObjects/WebElement';
+import { generateRandomId, generateRandomNumericId } from '../../support/utils';
+
+const openContentTemplatesPage = () => {
+  cy.visit('/');
+  let currentPage = new HomePage();
+  currentPage     = currentPage.getMenu().getContent().open();
+  return currentPage.openTemplates();
+};
+
+const addContentTemplate = template => cy.contentTemplatesController().then(controller => controller.addContentTemplate(template));
+const deleteContentTemplate = id => cy.contentTemplatesController().then(controller => controller.deleteContentTemplate(id));
+
+describe('Content Templates', () => {
+
+  let currentPage;
+  let template = {
+    contentType: 'BNR',
+    contentTypeText: 'Banner',
+    contentShape: '<div>test</div>'
+  };
+
+  beforeEach(() => {
+    template.id = generateRandomNumericId();
+    template.descr = generateRandomId();
+
+    cy.kcLogin('admin').as('tokens');
+  });
+
+  afterEach(() => cy.kcLogout());
```

Comment:
Try to move the teardown operations (i.e. deletion of the template) in the after each, to make sure it's always deleted even if the test fails